File: 69-Sqrtx/69-Sqrtx.java
// Last updated: 8/11/2026, 12:26:12 PM
class Solution {
    public int mySqrt(int x) {
        if (x == 0 || x == 1) return x;

        int i = 1;

        
        while ((long) i * i <= x) {
            if((i+1)*(i+1)>x){
                return i;
            }
            i++;
        }

        return i - 1;
    }
}

File: 69-Sqrtx/69-Sqrtx.py
# Last updated: 9/2/2026, 12:43:49 PM
class Solution:
    def mySqrt(self, x):
        left = 0
        right = x
        answer = 0

        while left <= right:
            mid = (left + right) // 2

            if mid * mid <= x:
                answer = mid
                left = mid + 1
            else:
                right = mid - 1

        return answer